File: spoj/codeforces/631_2/1.py
for _ in range(int(input())):
    n,x = list(map(int,input().split()))
    arr = list(map(int,input().split()))
    h = [0]*500
    for val in arr:
        h[val]+= 1
    ans = None
    for i in range(1,len(h)):
        val = h[i]
#         print(i,end = " ") 
        if val == 0:
            x -= 1
        else:
            continue
        if x < 0:
            ans = i-1
            break
    print(ans)

File: spoj/codeforces/648_2/1.cpp
#include<bits/stdc++.h>
using namespace std;

void print(int arr[],int n){
    for(int i=0;i<n;i++)cout<<arr[i]<<" ";
    cout<<endl;
}
int main(){

    int t;
    cin>>t;
    while(t--){
        int n,m;cin>>n>>m;
        int pp = max(n,m);
        int row[pp] = {0};
        int col[pp] = {0};
        for(int i=0;i<n;i++){
            for(int j=0;j<m;j++){
                int x;cin>>x;
                if(x == 1){
                    row[j] = 1;
                    col[i] = 1;
                }
            }
        }
//         print(row,pp);
        int ans = 0;
        for(int i=0;i<pp;i++)if(row[i] == 0 and col[i] == 0)ans++;
        cout<<ans<<endl;
        if(ans %2 ==0  ){
            cout<<"Vivek\n";
        }
        else cout<<"Ashish\n";
        

    }
}
